addons/mail/static/src/components/notification_list/notification_list.js

```diff
@@ -104,13 +104,13 @@ class NotificationList extends Component {
                 if (t1.localMessageUnreadCounter === 0 && t2.localMessageUnreadCounter > 0) {
                     return 1;
                 }
-                if (t1.lastMessage && t2.lastMessage) {
-                    return t1.lastMessage.date.isBefore(t2.lastMessage.date) ? 1 : -1;
+                if (t1.lastNonTransientMessage && t2.lastNonTransientMessage) {
```

Comment:
That's because we remove transient messages from the equation. Basically the separator show himself under the last message seen and as a transient message cannot be "seen" the separator can never be displayed somewhere else than under a message. So we need to add back these transient messages somehow in the equations. To ease all the process including server, I would suggest to have a boolean "lastMessageSeenByCurrentPartnerIsTransient" and a "lastSeenTransientMessage" and based on that show the separator at the right place.

The other solution is to accept that a transient message can be locally seen but prevent it from being told to server.
What approach would you prefer ?

Reply:
Added a skipped test that fails for now, need to make the decision to determine how to address this

Reply:
We probably need `lastSeenByCurrentPartnerMessageId` (same as now) and `lastSeenByCurrentPartnerNonTransientMessageId`, use the generic in the xml (same as before PR) and use the "non transient" one in the rest of the code (especially send to server).

Reply:
Best would be if separator was in model and we could decide where to place it depending on the following messages (we want to place it after the last seen message that is not a transient one).

Reply:
If this would be possible with your suggestion of `lastMessageSeenByCurrentPartnerIsTransient` feel free to try, I'm not sure what it implies, just do the one you think is best that works and we'll see from there.

Reply:
Seems to work like this. If it seems ok I'll retarget